packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -196,7 +196,22 @@ class SliverConstraints extends Constraints {
   /// the earliest visible part of this sliver in the [AxisDirection].
   ///
   /// For example, if [AxisDirection] is [AxisDirection.down], then this is the
-  /// scroll offset at the top of the visible portion of the sliver.
+  /// amount the top of the visible portion of the sliver has been scrolled
+  /// past the top of the viewport.
+  ///
+  /// This value is typically used to compute whether this sliver should still
+  /// protrude into the viewport via [SliverGeometry.paintExtent] and
+  /// [SliverGeometry.layoutExtent] considering how far the beginning of the
+  /// sliver is above the beginning of the viewport.
+  ///
+  /// For slivers past the end of the viewport, viewports typically stops trying
+  /// to compute the [scrollOffset]s of slivers that are in the 'future'
```

Comment:
it doesn't stop trying to compute it, it's just that the scroll offset at the top of the visible portion of the sliver is zero for those slivers.

Reply:
'future' direction -> "in the growth direction"?